### PR TITLE
Update `ApmStatsQuery` column interface parsing

### PR DESCRIPTION
**Pull request by anbnyc**
`ApmStatsQuery` column object needs to be built/parsed property by property, instead of casting as `ApmStatsQueryColumnType`.

### CODE REVIEW

File: datadog/resource_datadog_dashboard.go [hunk 1]
```diff
@@ -5181,6 +5181,26 @@ func getApmStatsQuerySchema() *schema.Schema {
...
+
+	datadogColumn := datadogV1.NewApmStatsQueryColumnTypeWithDefaults()
+
+	if value, ok := terraformColumn["name"].(string); ok && len(value) != 0 {
+		datadogColumn.SetName(value)
+	}
+	if value, ok := terraformColumn["alias"].(string); ok && len(value) != 0 {
+		datadogColumn.SetAlias(value)
+	}
+	if value, ok := terraformColumn["cell_display_mode"].(datadogV1.TableWidgetCellDisplayMode); ok && len(value) != 0 {
```
Comment (nmuesch) [inline datadog/resource_datadog_dashboard.go:5194]: You may have to build these as well?

**Review by nmuesch — COMMENTED** [review]
Overall looks good to me! Can you add to the dashboard tests to include a widget that includes this apm stats column?

File: datadog/resource_datadog_dashboard.go (hunk 1 above)
Comment (anbnyc) [reply datadog/resource_datadog_dashboard.go:5194]: @nmuesch `TableWidgetCellDisplayMode` and `WidgetSort` are enums, so I don't think there's anything to build --would it be better to set them as `.(string)` here?

Comment (anbnyc) [reply datadog/resource_datadog_dashboard.go:5194]: (Update: I tried `.(string)` and it throws a type error)

Comment (therve) [reply datadog/resource_datadog_dashboard.go:5194]: You need to cast to string, and then call `datadogV1.TableWidgetCellDisplayMode` on the value.

File: datadog/resource_datadog_dashboard.go [hunk 2]
```diff
@@ -5181,6 +5181,26 @@ func getApmStatsQuerySchema() *schema.Schema {
...
+		datadogColumn.SetCellDisplayMode(value)
+	}
+	if value, ok := terraformColumn["order"].(datadogV1.WidgetSort); ok && len(value) != 0 {
```
Comment (therve) [inline datadog/resource_datadog_dashboard.go:5197]: Same here.

**Review by therve — CHANGES_REQUESTED** [review]
I think we have an issue with the spec, it shouldn't set the default value of `cell_display_mode`.

**Review by therve — APPROVED** [review]

**Pull request merged**